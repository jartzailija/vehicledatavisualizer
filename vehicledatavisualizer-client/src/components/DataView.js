import React from 'react';
import { Row, Table, Col } from 'reactstrap';
import PropTypes from 'prop-types';
import styled from 'styled-components';

const DataView = ({hasData, municipality, carBrand, age, mileage, count, motorSize, power}) => {

  const HeaderContainer = styled.div`
    margin-top: 20px;
  `;

  const MarginRow = styled(Row)`
    margin: 20px;
  `

  const municipalityStr = municipality === '' ? '' : `in ${municipality}`;

  if(hasData) {
    return(
      <Row>
        <Col md={11}>
          <HeaderContainer>
            <h2>Data about {carBrand}s {municipalityStr}</h2>
          </HeaderContainer>
          <Table size="sm">
            <thead>
              <tr>
                <th>Count</th>
                <th>Avg. age</th>
                <th>Avg. mileage</th>
                <th>Avg. power</th>
                <th>Avg. cylinder size</th>
              </tr>
            </thead>
            <tbody>
              <tr>
                <td>{count}</td>
                <td>{age} yrs</td>
                <td>{mileage} km</td>
                <td>{power} hp</td>
                <td>{motorSize} cc</td>
              </tr>
            </tbody>
          </Table>
        </Col>
      </Row>
    );
  }
  else {
    return(<MarginRow>Select a sector in the chart</MarginRow>);
  }
};

DataView.propTypes = {
  hasData: PropTypes.bool.isRequired,
  age: PropTypes.number.isRequired,
  mileage: PropTypes.number.isRequired,
  count: PropTypes.number.isRequired,
  motorSize: PropTypes.number.isRequired,
  power: PropTypes.number.isRequired,
  municipality: PropTypes.string.isRequired,
  carBrand: PropTypes.string.isRequired
};

export default DataView;