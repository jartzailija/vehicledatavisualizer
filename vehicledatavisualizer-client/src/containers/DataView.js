import {
  connect
} from 'react-redux';
import DataView from '../components/DataView';

const mapStateToProps = state => ({
  hasData: state.averageData.hasData,
  age: state.averageData.age,
  mileage: state.averageData.mileage,
  count: state.averageData.count,
  motorSize: state.averageData.motorSize,
  power: state.averageData.power,
  carBrand: state.carBrands.selected,
  municipality: state.municipalities.selected
});


const ConnectDataView = connect(
  mapStateToProps
)(DataView);

export default ConnectDataView;
  