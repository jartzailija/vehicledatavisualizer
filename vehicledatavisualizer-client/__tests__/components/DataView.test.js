import React from 'react';
import { shallow } from 'enzyme';
import toJson from 'enzyme-to-json';

// Component to be tested
import DataView from '../../src/components/DataView';

describe('<DataView />', () => {
  describe('render()', () => {
    test('renders the component with data', () => {
      const params = { 
        hasData: true,
        age: 13.6,
        motorSize: 1730,
        mileage: 204759,
        power: 82,
        municipality: 'TestMunicipality',
        count: 123,
        carBrand: "Toyota"
      }
      const wrapper = shallow(<DataView {...params} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });

    test('renders the component without data', () => {
      const params = { 
        hasData: false,
        age: 0,
        motorSize: 0,
        mileage: 0,
        power: 0,
        municipality: '',
        count: 0,
        carBrand: ""
      }
      const wrapper = shallow(<DataView {...params} />);
      const component = wrapper.dive();

      expect(toJson(component)).toMatchSnapshot();
    });
  });
});