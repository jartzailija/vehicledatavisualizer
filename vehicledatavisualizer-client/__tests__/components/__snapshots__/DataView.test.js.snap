// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DataView /> render() renders the component with data 1`] = `
<div
  className="row"
>
  <Col
    md={11}
    tag="div"
    widths={
      Array [
        "xs",
        "sm",
        "md",
        "lg",
        "xl",
      ]
    }
  >
    <styled.div>
      <h2>
        Data about 
        Toyota
        s 
        in TestMunicipality
      </h2>
    </styled.div>
    <Table
      responsiveTag="div"
      size="sm"
      tag="table"
    >
      <thead>
        <tr>
          <th>
            Count
          </th>
          <th>
            Avg. age
          </th>
          <th>
            Avg. mileage
          </th>
          <th>
            Avg. power
          </th>
          <th>
            Avg. cylinder size
          </th>
        </tr>
      </thead>
      <tbody>
        <tr>
          <td>
            123
          </td>
          <td>
            13.6
             yrs
          </td>
          <td>
            204759
             km
          </td>
          <td>
            82
             hp
          </td>
          <td>
            1730
             cc
          </td>
        </tr>
      </tbody>
    </Table>
  </Col>
</div>
`;

exports[`<DataView /> render() renders the component without data 1`] = `
<StyledComponent
  forwardedComponent={
    Object {
      "$$typeof": Symbol(react.forward_ref),
      "attrs": Array [],
      "componentStyle": ComponentStyle {
        "componentId": "sc-bxivhb",
        "isStatic": true,
        "rules": Array [
          "
    margin: 20px;
  ",
        ],
      },
      "displayName": "Styled(Row)",
      "foldedComponentIds": Array [],
      "render": [Function],
      "styledComponentId": "sc-bxivhb",
      "target": [Function],
      "toString": [Function],
      "warnTooManyClasses": [Function],
      "withComponent": [Function],
    }
  }
  forwardedRef={null}
>
  Select a sector in the chart
</StyledComponent>
`;
